qutip/tensor.py

```diff
@@ -328,6 +300,8 @@ def tensor_swap(q_oper, *pairs):
     return Qobj(inpt=data, dims=dims, superrep=q_oper.superrep)
 
 
+import qutip.states
+
 def tensor_contract(qobj, *pairs):
```

Comment:
Just FYI: This whole routine can be made much faster by moving the loops to Cython.

Reply:
noted and appended to todo